nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/FileSystemRepository.java

```diff
@@ -199,13 +199,13 @@ public FileSystemRepository(final NiFiProperties nifiProperties) throws IOExcept
             for (final Map.Entry<String, Path> container : containers.entrySet()) {
                 final String containerName = container.getKey();
 
-                final long capacity = Files.getFileStore(container.getValue()).getTotalSpace();
+                final long capacity = container.getValue().toFile().getTotalSpace();
```

Comment:
I agree with you that exception should be thrown. I would prefer a RuntimeException with a message like

>  System returned total space of the partition for {reponame} is zero byte. Nifi can not create a zero sized FileSystemRepository

As Javadoc says that IOException signals that an I/O exception of some sort has occurred. This class is the general class of exceptions produced by failed or interrupted I/O operations.